src/Microsoft.AspNetCore.SignalR.Client/HubConnectionBuilderHttpExtensions.cs

```diff
@@ -2,283 +2,42 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Collections.ObjectModel;
-using System.Net;
-using System.Net.Http;
-using System.Net.WebSockets;
-using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Connections;
 using Microsoft.AspNetCore.Http.Connections;
 using Microsoft.AspNetCore.Http.Connections.Client;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 
 namespace Microsoft.AspNetCore.SignalR.Client
 {
     public static class HubConnectionBuilderHttpExtensions
     {
-        public static readonly string TransportTypeKey = "TransportType";
-        public static readonly string HttpMessageHandlerKey = "HttpMessageHandler";
-        public static readonly string HeadersKey = "Headers";
-        public static readonly string AccessTokenFactoryKey = "AccessTokenFactory";
-        public static readonly string WebSocketOptionsKey = "WebSocketOptions";
-        public static readonly string CookiesKey = "Cookies";
-        public static readonly string ProxyKey = "Proxy";
-        public static readonly string ClientCertificatesKey = "ClientCertificates";
-        public static readonly string CredentialsKey = "Credentials";
-        public static readonly string UseDefaultCredentialsKey = "UseDefaultCredentials";
-
-        public static IHubConnectionBuilder WithUrl(this IHubConnectionBuilder hubConnectionBuilder, string url)
+        public static IHubConnectionBuilder WithHttpConnection(this IHubConnectionBuilder hubConnectionBuilder, Action<HttpConnectionOptions> configureHttpConnection)
```

Comment:
Make the `url` a first class argument. It's the only mandatory setting, so it should be "required" by the API

Reply:
Sounds good. That will also allow an overload of string and Uri.